io/private/las/Utils.cpp

```diff
@@ -682,11 +682,27 @@ void V14BaseLoader::load(PointRef& point, const char *buf, int bufsize)
 void V14BaseLoader::pack(const PointRef& point, char *buf, int bufsize)
 {
     LeInserter ostream(buf, bufsize);
-    int32_t xi = (int32_t)m_scaling.m_xXform.toScaled(point.getFieldAs<double>(Dimension::Id::X));
-    int32_t yi = (int32_t)m_scaling.m_yXform.toScaled(point.getFieldAs<double>(Dimension::Id::Y));
-    int32_t zi = (int32_t)m_scaling.m_zXform.toScaled(point.getFieldAs<double>(Dimension::Id::Z));
 
-    ostream << xi << yi << zi;
+    auto converter = [](double val, Dimension::Id dim) -> int32_t
```

Comment:
I don't understand this change.

The change in #4088 was for a bad header value. The stored point value was correct. Can you explain?

Reply:
If reading a LAS file, it gets decoded from an integer to float via `offset + scale * (double)encodedValue`. When writing we do `(coordinate - offset) / scale`, and we should get the same encoded integer value back as we started with. But because the code truncates instead of rounds 

Currently it casts, but that means, for example:

```
offset = 1000000
scale = 0.001
encoded coordinate = 406018497

coordinate = 1000000 + 0.001 * (double)406018497 == 1406018.497

encoded = (1406018.497 - 1000000) / 0.001 == 406018496.99999994

int(406018496.99999994) == 406018496
round(406018496.99999994) == 406018497
```

Impact? `pdal translate input.las output.copc.laz` ends up with different encoded coordinate values. And when they're subsequently decoded the coordinate has moved.

Reply:
Implementation-wise, I would have used `std::round()` but the code already in V10BaseLoader (and missing from V14BaseLoader) achieved the same thing.